protocol/daemons/slinky/client/price_fetcher.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"cosmossdk.io/log"
+	"google.golang.org/grpc"
+
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+	"github.com/skip-mev/slinky/service/servers/oracle/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/pricefeed/api"
+	daemontypes "github.com/dydxprotocol/v4-chain/protocol/daemons/types"
+)
+
+// PriceFetcher is responsible for pulling prices from the slinky sidecar and sending them to the pricefeed server.
+type PriceFetcher interface {
+	Start(ctx context.Context) error
+	Stop()
+	FetchPrices(ctx context.Context) error
+}
+
+// PriceFetcherImpl implements the PriceFetcher interface.
+type PriceFetcherImpl struct {
+	marketPairFetcher      MarketPairFetcher
+	priceFeedServiceClient api.PriceFeedServiceClient
+	grpcClient             daemontypes.GrpcClient
+	pricesSocket           string
+	pricesConn             *grpc.ClientConn
+	slinky                 oracleclient.OracleClient
+	logger                 log.Logger
+}
+
+// NewPriceFetcher creates a PriceFetcher.
+func NewPriceFetcher(
+	marketPairFetcher MarketPairFetcher,
+	grpcClient daemontypes.GrpcClient,
+	pricesSocket string,
+	slinky oracleclient.OracleClient,
+	logger log.Logger) PriceFetcher {
+	return &PriceFetcherImpl{
+		marketPairFetcher: marketPairFetcher,
+		grpcClient:        grpcClient,
+		pricesSocket:      pricesSocket,
+		slinky:            slinky,
+		logger:            logger,
+	}
+}
+
+// Start initializes the underlying connections of the PriceFetcher.
+func (p *PriceFetcherImpl) Start(ctx context.Context) error {
+	cancelCtx, cf := context.WithTimeout(ctx, SlinkyPriceServerConnectionTimeout)
+	defer cf()
+	pricesConn, err := p.grpcClient.NewGrpcConnection(cancelCtx, p.pricesSocket)
+	if err != nil {
+		return err
+	}
+	p.pricesConn = pricesConn
+	p.priceFeedServiceClient = api.NewPriceFeedServiceClient(p.pricesConn)
+	return p.slinky.Start(ctx)
+}
+
+// Stop closes all open connections.
+func (p *PriceFetcherImpl) Stop() {
```

Comment:
wb priceFeedServiceClient + grpcClient connections

Reply:
The priceFeedServiceClient is just an interface wrapping the underlying grpc connection which is the pricesConn.

The grpcClient facilitates closing the pricesConn. 

Maybe it's equivalent to just calling `pricesConn.Close()`?